### PR TITLE
Add and set the new VolumeSnapshotHandlePairList field

### PR DESCRIPTION
**Pull request by leonardoce**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
/kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

Adds and sets the new field VolumeSnapshotHandlePairList of the VolumeGroupSnapshotContent resource, as described in the [latest iteration of the relative KEP](https://github.com/kubernetes/enhancements/tree/e1d4725d07a07c40da7ded6e1dd44c0f9aaeaeb9/keps/sig-storage/3476-volume-group-snapshot#volumegroupsnapshotcontent).

This is a prerequisite to promote the volume group snapshot API to v1beta1.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

This has been extracted from https://github.com/kubernetes-csi/external-snapshotter/pull/1150

This is an example of how a VolumeGroupSnapshotContent looks:

```
❯ kubectl get volumegroupsnapshotcontent groupsnapcontent-cea7182b-4b04-4614-a3e8-a5b09c669e5c  -o yaml
apiVersion: groupsnapshot.storage.k8s.io/v1alpha1
kind: VolumeGroupSnapshotContent
metadata:
  creationTimestamp: "2024-10-23T15:24:11Z"
  finalizers:
  - groupsnapshot.storage.kubernetes.io/volumegroupsnapshotcontent-bound-protection
  generation: 1
  name: groupsnapcontent-cea7182b-4b04-4614-a3e8-a5b09c669e5c
  resourceVersion: "458156"
  uid: 740d4c07-9efc-4c6b-b46a-3c4733a391d3
spec:
  deletionPolicy: Delete
  driver: hostpath.csi.k8s.io
  source:
    volumeHandles:
    - 75b1b2e9-8cc4-11ef-b606-c20bfb388315
    - 75b196d5-8cc4-11ef-b606-c20bfb388315
  volumeGroupSnapshotClassName: csi-hostpath-groupsnapclass
  volumeGroupSnapshotRef:
    apiVersion: groupsnapshot.storage.k8s.io/v1alpha1
    kind: VolumeGroupSnapshot
    name: new-groupsnapshot-demo
    namespace: default
    resourceVersion: "458124"
    uid: cea7182b-4b04-4614-a3e8-a5b09c669e5c
status:
  creationTime: 1729697051387803433
  pvVolumeSnapshotContentList:
  - persistentVolumeRef:
      name: pvc-4469b2c3-c5ca-415e-a601-427de7ebe333
    volumeSnapshotContentRef:
      name: snapcontent-5943c2815c1b999992c542a608e005e211cbedaf8165b06b4a33a96876a25752-2024-10-23-3.24.13
  - persistentVolumeRef:
      name: pvc-7cd5e901-3c7e-464f-9bff-2adb860487e6
    volumeSnapshotContentRef:
      name: snapcontent-d12702e75e2f70c6437ff30683d8e76fb6d2b0f8e8d08b950d30ca919242e063-2024-10-23-3.24.13
  readyToUse: true
  volumeGroupSnapshotHandle: daacc21b-9152-11ef-b66d-2e29031d8e65
  volumeSnapshotHandlePairList:
  - snapshotHandle: daaccc26-9152-11ef-b66d-2e29031d8e65
    volumeHandle: 75b1b2e9-8cc4-11ef-b606-c20bfb388315
  - snapshotHandle: db2e0c02-9152-11ef-b66d-2e29031d8e65
    volumeHandle: 75b196d5-8cc4-11ef-b606-c20bfb388315
```

**Does this PR introduce a user-facing change?**:

```release-note
Add a field called `volumegroupsnapshotcontent.status.volumeSnapshotHandlePairList` that allows the consumer to quickly map volume handles with snapshot handles.
```

### CODE REVIEW

**Comment by xing-yang** [comment]
Can you please fix the PR title?  It is VolumeSnapshotHandlePairList, not pvVolumeSnapshotContentList.

File: client/apis/volumegroupsnapshot/v1alpha1/types.go [hunk 1]
```diff
@@ -333,6 +333,12 @@ type VolumeGroupSnapshotContentStatus struct {
 	// +optional
 	VolumeGroupSnapshotHandle *string `json:"volumeGroupSnapshotHandle,omitempty" protobuf:"bytes,1,opt,name=volumeGroupSnapshotHandle"`
 
+	// VolumeSnapshotHandlePairList is a list of CSI "volume_id" and "snapshot_id"
+	// pair returned by the CSI driver to identify snapshots and their source volumes
+	// on the storage system.
+	// +optional
+	VolumeSnapshotHandlePairList []VolumeSnapshotHandlePair `json:"volumeSnapshotHandlePairList,omitempty" protobuf:"bytes,6,opt,name=volumeSnapshotHandlePairList"`
```
Comment (xing-yang) [inline client/apis/volumegroupsnapshot/v1alpha1/types.go:340]: Can you move this to line 368 after item 5?  It's confusing to see it jump from 1 to 6.

Comment (leonardoce) [reply client/apis/volumegroupsnapshot/v1alpha1/types.go:340]: I followed the order specified in the KEP, but having it after item 5 is definitely better

**Comment by leonardoce** [comment]
@xing-yang I just applied your suggestions. Thanks!

**Comment by jsafrane** [comment]
I take it as a first step towards API rework. I would expect some pairs to be removed in the end, like  `VolumeGroupSnapshotContentStatus.PVVolumeSnapshotContentList` is actually replaced by the new field and perhaps `VolumeGroupSnapshotStatus.PVCVolumeSnapshotRefList` is useless too.

**Comment by xing-yang** [comment]
> I take it as a first step towards API rework. I would expect some pairs to be removed in the end, like VolumeGroupSnapshotContentStatus.PVVolumeSnapshotContentList is actually replaced by the new field and perhaps VolumeGroupSnapshotStatus.PVCVolumeSnapshotRefList is useless too.
@jsafrane VolumeGroupSnapshotStatus.PVCVolumeSnapshotRefList is still required, but we will try to remove VolumeGroupSnapshotContentStatus.PVVolumeSnapshotContentList at a later phase. We have the plan listed in this doc: https://docs.google.com/document/d/1jjliw0tFhCPnT9ixlYpj61k5u8_-awf1qSCWrkunshQ/edit?tab=t.0#heading=h.fizenae5evi0

**Comment by andyzhangx** [comment]
that's a transient error, rerun should work

**Pull request merged**